core/src/main/scala/org/apache/spark/util/ListenerBus.scala

```diff
@@ -80,7 +89,16 @@ private[spark] trait ListenerBus[L <: AnyRef, E] extends Logging {
       }
       try {
         doPostEvent(listener, event)
+        if (Thread.interrupted()) {
+          logError(s"Interrupted while posting to ${Utils.getFormattedClassName(listener)}.  " +
+            s"Removing that listener.")
+          removeListenerOnError(listener)
```

Comment:
if the thread state is interrupted, is it still safe to keep this queue running?

Reply:
You could just throw `InterruptedException` here to avoid duplicating the error handling code.

Reply:
+1